selftests/unit/test_vm.py

```diff
@@ -1,71 +1,69 @@
-        mock_vm.should_receive('create_snapshot').once().ordered()
-        # VMTestRunner()
-        Args = flexmock(test_result_total=1,
-                        url=['/tests/sleeptest', '/tests/other/test',
-                             'passtest'],
-                        vm_domain='domain',
-                        vm_username='username',
-                        vm_hostname='hostname',
-                        vm_port=22,
-                        vm_password='password',
-                        vm_key_file=None,
-                        vm_cleanup=True,
-                        vm_no_copy=False,
-                        vm_timeout=120,
-                        vm_hypervisor_uri='my_hypervisor_uri',
-                        env_keep=None)
-        log = flexmock()
-        log.should_receive("info")
-        job = flexmock(args=Args, log=log)
-        self.runner = avocado_runner_vm.VMTestRunner(job, None)
-        mock_vm.should_receive('stop').once().ordered()
-        mock_vm.should_receive('restore_snapshot').once().ordered()
-
-    def tearDown(self):
-        flexmock_teardown()
-
     def test_setup(self):
-        """ Tests VMTestRunner.setup() """
-        self.runner.setup()
-        self.runner.tear_down()
-        flexmock_teardown()
+        mock_vm = _FakeVM()
+        mock_vm.start = mock.Mock(return_value=True)
+        mock_vm.create_snapshot = mock.Mock()
+        mock_vm.stop = mock.Mock()
+        mock_vm.restore_snapshot = mock.Mock()
+        job_args = argparse.Namespace(test_result_total=1,
+                                      vm_domain='domain',
+                                      vm_username='username',
+                                      vm_hostname='hostname',
+                                      vm_port=22,
+                                      vm_password='password',
+                                      vm_key_file=None,
+                                      vm_cleanup=True,
+                                      vm_no_copy=False,
+                                      vm_timeout=120,
+                                      vm_hypervisor_uri='my_hypervisor_uri',
+                                      reference=['/tests/sleeptest.py',
+                                                 '/tests/other/test',
+                                                 'passtest.py'],
+                                      dry_run=True,
+                                      env_keep=None)
+        try:
+            job = Job(job_args)
+            with mock.patch('avocado_runner_vm.vm_connect',
+                            return_value=mock_vm):
+                # VMTestRunner()
+                runner = avocado_runner_vm.VMTestRunner(job, None)
+                runner.setup()
+                runner.tear_down()
+                mock_vm.start.assert_called_once_with()
+                mock_vm.create_snapshot.assert_called_once_with()
+                mock_vm.stop.assert_called_once_with()
```

Comment:
it'd be better to check for the calls before the `tear_down` (I mean check that `start` and `create_snapshot` was called after `setup` and `stop` and `restore` after `tear_down`). But again, it's really minor so take it or let it as is.

Reply:
Your point is valid.  On the other hand, having it checked *after* `tear_down` has another interesting side effect: it makes sure that `tear_down` is not touching places it shouldn't (because of the called_**once**_with()).  Humm... this is a tough one.  When in doubt, let's leave it as is.